IPs/RSKIP243.md

```diff
@@ -0,0 +1,77 @@
+
+# **Motivation**
+
+The motivation for the removal of storage refunds can be found in [EIP3298]([https://eips.ethereum.org/EIPS/eip-3298). Essentially, gas refunds that span over blocks bring numerous problems. However, instead of removing storage gas reimbursements completely, we propose keeping them but only if the storage creation and removal occurs within a single transaction. By doing so do not harm common Solidity language constructions, such as reentrancy semaphores. This change keeps the platform fair because on refunds the trie is not impacted, and the I/O cost is avoided. 
+
+Also currently when a contract is destroyed, the reimbursed amount is constant. However, the variable cost related to the number of bytes installed can be as high as 4915200 gas. We propose to reimburse this amount, increasing platform fairness.
+
+Because intra-transaction refunds cannot increase block processing time, the cap to 50% of the consumed gas is removed.
+
+
+# **Specification**
+
+Starting from an activation block (TBD) storage cleanup reimbursements are modified both for `SSTORE` and `SELFDESTRUCT`. Refunds can only be collected in the same transaction they the gas is spent. 
+
+Two data structures are created by the virtual machine when processing a transaction: a set *contracts_created\[contractAddress\]* (one for all contracts) and one map of sets *storage_cells_created\[contractAddress\]\[storageAddress\]*. Both are empty on creation.
+
+If a contract is created, it is added to *contracts_created*. If a contract is destroyed, the contract address is looked up in *contracts_created*. If it exists, then 24K gas plus the refund of the variable cost required to install the code is refunded at the end of transaction processing. 
+
+If a storage cell is created (zero value before) its address is added to the set *storage_cells_created\[contractAddress\]*, where contractAddress is the current contract. If in the same transaction the same cell address is cleared (zero value afterward), a refund of (*creationCost* -3000) is added to the refund amount, and the storage cell address is removed from *storage_cells_created\[contractAddress\]*. *creationCost* may be the 20K established at RSK genesis, or a new cost established by other RSKIP. If *creationCost* is variable, then the reimbursement will also be variable to match *creationCost* .
+
+If a contract call revers, the changes on *contracts_created* and *storage_cells_created* are not reverted. This implies that clearing a storage cell may trigger a refund even if the cell was already cleared. 
+
+The cap of the refund to 50% of the transaction gas consumed is removed.
+
+
+# Rationale
+
+Several related topics are analyzed in separate sections.
+
+### This proposal vs. [EIP2929](https://eips.ethereum.org/EIPS/eip-2929)
```

Comment:
Do you mean EIP-3298 (instead of 2929)? There are some connections to EIP-2929, but those are not direct. EIP-2929 was motivated primarily as DoS mitigation.

Reply:
No. I'm really referring to the part of EIP2929 that proposes the access lists., not the part that changes the opcode costs. Access lists may at first sight look similar to creation lists, but serve a completely different purposes.
Access lists change cost to take into account intra-block caches in the node (read operations).
Creation lists change cost to take into account the case where nothing is written to SSD in the node. (write operations)

Reply:
I'm pushing a new version with hopefully a more clear section.

Reply:
@smishraIOV Tell me if the clarification is satisfactory for merging.